### PR TITLE
Render scale update related support and fixes

### PR DESCRIPTION
**Pull request by moudgils**
## What does this PR do?

- Add support to do runtime r_renderScale updates
- Fix 2d debug text rendering to always stay on the top right side even if the render resolution scale changes

## How was this PR tested?

Tested on PC and ios

### CODE REVIEW

File: Gems/AtomLyIntegration/AtomViewportDisplayInfo/Code/Source/AtomViewportDisplayInfoSystemComponent.cpp [hunk 1]
```diff
@@ -163,8 +163,12 @@ namespace AZ::Render
...
 
         m_drawParams.m_drawViewportId = viewportContext->GetId();
-        auto viewportSize = viewportContext->GetViewportSize();
-        m_drawParams.m_position = AZ::Vector3(static_cast<float>(viewportSize.m_width), 0.0f, 1.0f) + AZ::Vector3(r_topRightBorderPadding) * viewportContext->GetDpiScalingFactor();
+        
+        AZ::IConsole* console = AZ::Interface<AZ::IConsole>::Get();
+        uint32_t r_width;
+        console->GetCvarValue("r_width", r_width);
+
+        m_drawParams.m_position = AZ::Vector3(static_cast<float>(r_width), 0.0f, 1.0f) + AZ::Vector3(r_topRightBorderPadding) * viewportContext->GetDpiScalingFactor();
```
Comment (VickyAtAZ) [inline Gems/AtomLyIntegration/AtomViewportDisplayInfo/Code/Source/AtomViewportDisplayInfoSystemComponent.cpp]: Do you need to check if the r_resolutionMode is 1?
For the resolution mode which doesn't use r_width, this won't be correct.

**Review by VickyAtAZ — CHANGES_REQUESTED** [review]

File: Gems/Atom/Bootstrap/Code/Source/BootstrapSystemComponent.cpp [hunk 2]
```diff
@@ -656,7 +664,7 @@ namespace AZ
                     return nullptr;
                 }
             }
-
+
```
Comment (AMZN-alexpete) [inline Gems/Atom/Bootstrap/Code/Source/BootstrapSystemComponent.cpp]: nit: space snuck in
```suggestion

```

File: Gems/Atom/Bootstrap/Code/Source/BootstrapSystemComponent.cpp [hunk 3]
```diff
@@ -82,7 +82,15 @@ AZ_CVAR(uint32_t, r_width, 1920, nullptr, AZ::ConsoleFunctorFlags::DontReplicate
 AZ_CVAR(uint32_t, r_height, 1080, nullptr, AZ::ConsoleFunctorFlags::DontReplicate, "Starting window height in pixels.");
 AZ_CVAR(uint32_t, r_fullscreen, false, nullptr, AZ::ConsoleFunctorFlags::DontReplicate, "Starting fullscreen state.");
 AZ_CVAR(uint32_t, r_resolutionMode, 0, nullptr, AZ::ConsoleFunctorFlags::DontReplicate, "0: render resolution same as window client area size, 1: render resolution use the values specified by r_width and r_height");
-AZ_CVAR(float, r_renderScale, 1.0f, nullptr, AZ::ConsoleFunctorFlags::DontReplicate, "Scale to apply to the window resolution.");
+
+void cvar_r_renderScale_Changed(const float& newRenderScale)
+{
+    AzFramework::WindowSize newSize =
```
Comment (AMZN-alexpete) [inline Gems/Atom/Bootstrap/Code/Source/BootstrapSystemComponent.cpp]: I think we should add a check if newRenderScale is less than or equal to 0.0 and AZ_Error if so - the commandline version has a check for this

File: Gems/AtomLyIntegration/AtomViewportDisplayInfo/Code/Source/AtomViewportDisplayInfoSystemComponent.cpp (hunk 1 above)
Comment (akioCL) [reply Gems/AtomLyIntegration/AtomViewportDisplayInfo/Code/Source/AtomViewportDisplayInfoSystemComponent.cpp]: > Do you need to check if the r_resolutionMode is 1? For the resolution mode which doesn't use r_width, this won't be correct.

You need to use:
```
            AzFramework::WindowSize windowSize;
            AzFramework::WindowRequestBus::EventResult(
                windowSize,
                windowHandle,
                &AzFramework::WindowRequestBus::Events::GetRenderResolution);
```

That returns the proper resolution if you use the native window size or a custom resolution.

Comment (moudgils) [reply Gems/AtomLyIntegration/AtomViewportDisplayInfo/Code/Source/AtomViewportDisplayInfoSystemComponent.cpp]: GetRenderResolution doesnt work as that will give the width after render scale is applied which will push the text to the middle of the screen. I do need r_width but for now I have wrapped it around r_resolutionMode. For the case where r_resolutionMode is 0 we probably need some code to handle Editor differently than launcher but that can come later. This fixes text rendering for mobile as r_resolutionMode is enabled for them. For platforms where you are running a gamelauncher and r_resolutionMode is not enabled will still have the issue where renderscale can move the positioning of debug text but that is of much less concern atm.

**Review by VickyAtAZ — APPROVED** [review]

**Pull request merged**